src/identity/hooks/useAvatar.ts

```diff
@@ -1,22 +1,32 @@
 import { publicClient } from '../../network/client';
-import { useOnchainActionWithCache } from '../../utils/hooks/useOnchainActionWithCache';
-import { GetEnsAvatarReturnType, normalize } from 'viem/ens';
+import { type GetEnsAvatarReturnType, normalize } from 'viem/ens';
+import { useQuery } from '@tanstack/react-query';
 
-export const ensAvatarAction = (ensName: string) => async (): Promise<GetEnsAvatarReturnType> => {
-  try {
-    return await publicClient.getEnsAvatar({
-      name: normalize(ensName),
-    });
-  } catch (err) {
-    return null;
-  }
+export const ensAvatarAction = async (ensName: string): Promise<GetEnsAvatarReturnType> => {
+  return await publicClient.getEnsAvatar({
+    name: normalize(ensName),
+  });
 };
 
-export const useAvatar = (ensName: string) => {
+type Arguments = {
```

Comment:
Let's move these types into the `identity/types.ts` file, and let's give them a more verbose name.

`Arguments` -> `AvatarOptions`
`QueryOptions` -> `AvatarQueryOptions`

and let's do the same for Name types.